extensions/amp-analytics/0.1/events.js

```diff
@@ -414,6 +421,29 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @return {!Promise}
+   * @visibleForTesting
+   */
+  createReportReadyPromise_() {
+    const viewer = this.root.getViewer();
+    let resolver = null;
```

Comment:
It's a scope leak. Sometimes it makes sense to scope `resolve` outside of the promise. That communicates that a promise can be resolved several different ways. In this case the situation is simpler - only one resolution involved and thus it's better to scope it in. It's no different then defining a `let` variable inside the `if` if it's only used in that branch.